legend-engine-shared-core/src/main/java/org/finos/legend/engine/shared/core/operational/opentracing/TracedSupplier.java

```diff
@@ -0,0 +1,73 @@
+//  Copyright 2023 Goldman Sachs
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package org.finos.legend.engine.shared.core.operational.opentracing;
+
+import io.opentracing.Scope;
+import io.opentracing.Span;
+import io.opentracing.Tracer;
+import io.opentracing.util.GlobalTracer;
+
+import java.util.function.Supplier;
+
+/**
+ * Consider tracing the executor service rather than the supplier via opentracing-contrib/java-concurrent
+ */
+public class TracedSupplier<T> implements Supplier<T>
```

Comment:
Did you look at the classes inside package `io.opentracing.contrib.concurrent`.  They seem to already handle concurrency on tracing.  Like, you can wrap the threadpoolexecutor on `TracedExecutorService` to handle this  seamlessly for y ou.

Reply:
Thanks, this makes sense updated the PR